### PR TITLE
reference/configuration: add a guide about how to use dynamic config

### PR DESCRIPTION
**Pull request by rleungx**
# What is changed, added or deleted? (Required)
This PR adds a guide about how to use the dynamic config.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
N/A

### CODE REVIEW

**Comment by TomShawn** [comment]
@rleungx Please add this document to TOC and fix the CI failure, thanks!

File: reference/configuration/dynamic-config.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# 动态配置变更使用文档
+
+动态配置变更主要是支持包括 TiDB，TiKV，PD 在内的各组件的配置进行在线更新。用户可以通过利用动态配置变更对各组件进行性能调优而无需重启集群组件。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
动态配置变更支持在线更新包括 TiDB、TiKV、PD 在内的各组件配置。用户可以利用动态配置变更对各组件进行性能调优，无需重启集群组件。
```

File: reference/configuration/dynamic-config.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+## 开启动态配置变更
+
+4.0 版本默认开启该参数，可通过修改 TiDB，TiKV，PD 配置文件中的 `enable-dynamic-config = false` 关闭该功能。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
TiDB 4.0 版本默认开启动态配置变更。可在 TiDB、TiKV、PD 的配置文件中设置 `enable-dynamic-config = false` 来关闭该功能。
```

File: reference/configuration/dynamic-config.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+目前可通过 pd-ctl 对各组件（TiDB，TiKV，PD）进行修改和查看。
+
+### 查看组件实例
+
+可通过 `component ids <component>` 查看组件实例。其中 `component` 为组件类型，目前支持 `tidb`，`tikv`，`pd` 三种类型。
+
+示例如下：
+
+```bash
+>> component ids tikv
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]: 用户拷贝这条命令去执行的话，是否需要带上 `>>` 呢？

File: reference/configuration/dynamic-config.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    "127.0.0.1:20161",
+    "127.0.0.1:20162"
+]
+```
+
+### 修改配置
+
+#### 修改全局配置
+
+可通过 `component set <component> <key> <value>` 进行设置，其中 `component` 为组件类型，含义同上。`key` 为参数名称，`value` 为参数值。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
可通过 `component set <component> <key> <value>` 进行设置，其中 `component` 为组件类型，目前支持 `tidb`，`tikv`，`pd` 三种类型。`key` 为参数名称，`value` 为参数值。
```

File: reference/configuration/dynamic-config.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+[pessimistic-txn]
+  enabled = true
+  wait-for-lock-timeout = 1000
+  wake-up-delay-duration = 20
+...
+```
+
+### 删除实例配置
+
+可通过 `component delete <component ID>` 进行删除，其中 `component ID` 为实例的 IP 地址加端口，如 `127.0.0.1:20160`。可通过使用 `ids` 命令进行查看。主要用于下线节点后，将配置删除。通常该命令无需手动操作。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
可通过 `component delete <component ID>` 命令删除实例。其中 `component ID` 为实例的 IP 地址加端口，如 `127.0.0.1:20160`。可通过使用 `ids` 命令进行查看。

该命令主要用于在下线节点后删除配置。通常该命令无需手动操作。
```

File: reference/configuration/dynamic-config.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+## 支持的参数列表
+
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion

动态配置变更功能支持动态变更以下配置项。

```

File: reference/configuration/dynamic-config.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+### TiDB
+
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion

动态配置变更功能支持动态变更以下 TiDB 配置项。

```

File: reference/configuration/dynamic-config.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+| 参数 | 简介 |
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
| 参数 | 描述 |
```

File: reference/configuration/dynamic-config.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+| log.expensive-threshold | 输出 expensive 操作的行数阈值 |
+| check-mb4-value-in-utf8 | 开启检查 utf8mb4 字符的开关，如果开启此功能，字符集是 utf8，且在 utf8 插入 mb4 字符，系统将会报错 |
+| enable-streaming | 开启 coprocessor 的 streaming 获取数据模式 |
+| txn-local-latches.capacity | Hash 对应的 slot 数，会自动向上调整为 2 的指数倍。每个 slot 占 32 Bytes 内存。当写入数据的范围比较广时（如导数据），设置过小会导致变慢，性能下降 |
+| compatible-kill-query | 设置 KILL 语句的兼容性 |
+| treat-old-version-utf8-as-utf8mb4 | 将旧表中的 utf8 字符集当成 utf8mb4的开关 |
+| opentracing.enable | 是否开启 opentracing 功能 |
+| prepared-plan-cache.enable | 是否开启 prepare 语句的 plan cache |
+
+具体参数意义可参考 [TiDB 配置文件描述](https://github.com/pingcap/docs-cn/blob/master/reference/configuration/tidb-server/configuration-file.md)。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
更详细的参数描述可参考 [TiDB 配置文件描述](https://github.com/pingcap/docs-cn/blob/master/reference/configuration/tidb-server/configuration-file.md)。
```

File: reference/configuration/dynamic-config.md [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+
+### TiKV
+
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion

动态配置变更功能支持动态变更以下 TiKV 配置项。

```

File: reference/configuration/dynamic-config.md [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+| 参数 | 简介 |
+| --- | --- |
+| refresh-config-interval | 尝试更新配置的时间间隔 |
+| raftstore.sync-log | 数据、log 落盘是否 sync |
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion
| raftstore.sync-log | 是否同步数据和落盘的 log |
```

File: reference/configuration/dynamic-config.md [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+| {db-name}.{cf-name}.soft-pending-compaction-bytes-limit | pending compaction bytes 的软限制 |
+| {db-name}.{cf-name}.hard-pending-compaction-bytes-limit | pending compaction bytes 的硬限制 |
+| {db-name}.{cf-name}.titan.blob-run-mode | 处理 blob 文件的模式 |
+
+上述前缀为 {db-name} 或 {db-name}.{cf-name} 的参数是 rocksdb 相关的配置
+`db-name` 的取值可以为是 `rocksdb`, `raftdb`
+当 `db-name` 为 `rocksdb` 时，cf-name 的取值有: `defaultcf`，`writecf`，`lockcf`，`raftcf`
+当 `db-name` 为 `raftdb` 时，cf-name 的取值有: `defaultcf`
+
+具体参数意义可参考 [TiKV 配置文件描述](https://github.com/pingcap/docs-cn/blob/master/reference/configuration/tikv-server/configuration-file.md)。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]: 上面出现的参数，很多并不在 [TiKV 配置文件描述](https://github.com/pingcap/docs-cn/blob/master/reference/configuration/tikv-server/configuration-file.md)中，是否要在该配置文档添加这些 item 或者更改这句描述？

File: reference/configuration/dynamic-config.md [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+
+### PD
+
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]:
```suggestion

动态配置变更功能支持动态变更以下 PD 配置项。

```

File: reference/configuration/dynamic-config.md [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+| schedule.disable-replace-offline-replica | 用于关闭迁移 OfflineReplica 的特性 |
+| schedule.disable-make-up-replica | 用于关闭补充副本的特性 |
+| schedule.disable-remove-extra-replica | 用于关闭删除多余副本的特性 |
+| schedule.disable-location-replacement | 用于关闭隔离级别检查 |
+| replication.max-replicas | 用于设置副本的数量 |
+| replication.location-labels | 用于设置 TiKV 集群的拓扑信息 |
+| label-property | 标签相关的配置项 |
+| pd-server.use-region-storage | 开启独立的 region 存储 |
+
+具体参数意义可参考 [PD 配置文件描述](https://github.com/pingcap/docs-cn/blob/master/reference/configuration/pd-server/configuration-file.md)。
```
Comment (TomShawn) [inline reference/configuration/dynamic-config.md]: ditto

File: reference/configuration/dynamic-config.md (hunk 3 above)
Comment (lhy1024) [reply reference/configuration/dynamic-config.md]: 不需要

File: reference/configuration/dynamic-config.md [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+
+动态配置变更主要是支持包括 TiDB，TiKV，PD 在内的各组件的配置进行在线更新。用户可以通过利用动态配置变更对各组件进行性能调优而无需重启集群组件。
+
+## 开启动态配置变更
+
+4.0 版本默认开启该参数，可通过修改 TiDB，TiKV，PD 配置文件中的 `enable-dynamic-config = false` 关闭该功能。
+
+## 常用操作
+
+目前可通过 pd-ctl 对各组件（TiDB，TiKV，PD）进行修改和查看。
```
Comment (lhy1024) [inline reference/configuration/dynamic-config.md]: 也可以通过 sql 吧

Comment (rleungx) [reply reference/configuration/dynamic-config.md]: 目前还在开发中

**Pull request closed without merging**